$NetBSD: patch-test_Makefile.am,v 1.3 2024/01/13 04:25:39 riastradh Exp $

Fix cross-build again.  noinst_PROGRAMS does not work for programs that
are executed at build-time.  Maybe some day automake will sprout a
built-in way to do the right thing here, but so far it hasn't, to my
knowledge.

--- test/Makefile.am.orig	2023-12-02 21:10:00.000000000 +0000
+++ test/Makefile.am
@@ -38,11 +38,10 @@ EXTRA_DIST=$(check_SCRIPTS) math-tables.
 
 math.5c: $(TABLES)
 
-noinst_PROGRAMS = math-tables
-
-math_tables_SOURCES = math-tables.c
-
-math_tables_LIBS = -lm
+math-tables$(BUILD_EXEEXT): math-tables.o
+	$(CC_FOR_BUILD) -o $@ $(CFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) math-tables.o -lm
+math-tables.o: math-tables.c
+	$(CC_FOR_BUILD) -o $@ $(CPPFLAGS_FOR_BUILD) $(CFLAGS_FOR_BUILD) -c math-tables.c
 
 $(TABLES): math-tables$(BUILD_EXEEXT)
 	./math-tables$(BUILD_EXEEXT) > $(TABLES)
